source/connection_manager.c

```diff
@@ -365,6 +365,113 @@ static void s_aws_http_connection_manager_move_front_acquisition(
     aws_linked_list_push_back(output_list, node);
 }
 
+/*
+ * Encompasses all of the non-state-update work that needs to be done for various
+ * events:
+ *   manager release
+ *   connection release
+ *   connection acquire
+ *   connection_setup
+ *   connection_shutdown
+ *
+ * The work unit is built under the manager's lock, but then executed
+ * outside of it.
+ */
+struct aws_connection_management_work_unit {
```

Comment:
Hmmm, after our discussion, what about ```aws_connection_manager_transaction```? Thoughts? If I read work_unit as transaction throughout this commit, everything makes sense.

Reply:
Yeah, I like that better.